diff --git a/dist/changesets-cli.cjs.js b/dist/changesets-cli.cjs.js
index 9f0bd6c5f96011be225269a71365a878cb822bb7..c426ae7c7fb931172b714b8070bd635b2d8385b7 100644
--- a/dist/changesets-cli.cjs.js
+++ b/dist/changesets-cli.cjs.js
@@ -744,25 +744,35 @@ async function getPublishTool(cwd) {
   const pm = await packageManagerDetector.detect({
     cwd
   });
-  if (!pm || pm.name !== "pnpm") return {
+  if (!pm) return {
     name: "npm"
   };
-  try {
-    let result = await spawn__default["default"]("pnpm", ["--version"], {
-      cwd
-    });
-    let version = result.stdout.toString().trim();
-    let parsed = semverParse__default["default"](version);
-    return {
-      name: "pnpm",
-      shouldAddNoGitChecks: (parsed === null || parsed === void 0 ? void 0 : parsed.major) === undefined ? false : parsed.major >= 5
-    };
-  } catch (e) {
+  if (pm.name === "bun") {
     return {
-      name: "pnpm",
-      shouldAddNoGitChecks: false
+      name: "bun"
     };
   }
+  if (pm.name === "pnpm") {
+    try {
+      let result = await spawn__default["default"]("pnpm", ["--version"], {
+        cwd
+      });
+      let version = result.stdout.toString().trim();
+      let parsed = semverParse__default["default"](version);
+      return {
+        name: "pnpm",
+        shouldAddNoGitChecks: (parsed === null || parsed === void 0 ? void 0 : parsed.major) === undefined ? false : parsed.major >= 5
+      };
+    } catch (e) {
+      return {
+        name: "pnpm",
+        shouldAddNoGitChecks: false
+      };
+    }
+  }
+  return {
+    name: "npm"
+  };
 }
 async function getTokenIsRequired() {
   const {
@@ -898,6 +908,10 @@ async function internalPublish(packageJson, opts, twoFactorState) {
       env: Object.assign({}, process.env, envOverride),
       cwd: opts.cwd,
       stdio: ["inherit", "inherit", "pipe"]
+    }) : publishTool.name === "bun" ? spawn__default["default"]("bun", ["publish", ...publishFlags], {
+      env: Object.assign({}, process.env, envOverride),
+      cwd: opts.publishDir,
+      stdio: ["inherit", "inherit", "pipe"]
     }) : spawn__default["default"](publishTool.name, ["publish", opts.publishDir, ...publishFlags], {
       env: Object.assign({}, process.env, envOverride),
       stdio: ["inherit", "inherit", "pipe"]
@@ -942,6 +956,9 @@ async function internalPublish(packageJson, opts, twoFactorState) {
   } = publishTool.name === "pnpm" ? await spawn__default["default"]("pnpm", ["publish", ...publishFlags], {
     env: Object.assign({}, process.env, envOverride),
     cwd: opts.cwd
+  }) : publishTool.name === "bun" ? await spawn__default["default"]("bun", ["publish", ...publishFlags.filter(flag => flag !== "--json")], {
+    env: Object.assign({}, process.env, envOverride),
+    cwd: opts.publishDir
   }) : await spawn__default["default"](publishTool.name, ["publish", opts.publishDir, ...publishFlags], {
     env: Object.assign({}, process.env, envOverride)
   });
diff --git a/dist/changesets-cli.esm.js b/dist/changesets-cli.esm.js
index d9582bf53eaf55402a6bf96a39589af7854b8f23..a4755b55cbeaeba2e05f486c090cfdd8b5b9dd24 100644
--- a/dist/changesets-cli.esm.js
+++ b/dist/changesets-cli.esm.js
@@ -707,25 +707,35 @@ async function getPublishTool(cwd) {
   const pm = await detect({
     cwd
   });
-  if (!pm || pm.name !== "pnpm") return {
+  if (!pm) return {
     name: "npm"
   };
-  try {
-    let result = await spawn$1("pnpm", ["--version"], {
-      cwd
-    });
-    let version = result.stdout.toString().trim();
-    let parsed = semverParse(version);
-    return {
-      name: "pnpm",
-      shouldAddNoGitChecks: (parsed === null || parsed === void 0 ? void 0 : parsed.major) === undefined ? false : parsed.major >= 5
-    };
-  } catch (e) {
+  if (pm.name === "bun") {
     return {
-      name: "pnpm",
-      shouldAddNoGitChecks: false
+      name: "bun"
     };
   }
+  if (pm.name === "pnpm") {
+    try {
+      let result = await spawn$1("pnpm", ["--version"], {
+        cwd
+      });
+      let version = result.stdout.toString().trim();
+      let parsed = semverParse(version);
+      return {
+        name: "pnpm",
+        shouldAddNoGitChecks: (parsed === null || parsed === void 0 ? void 0 : parsed.major) === undefined ? false : parsed.major >= 5
+      };
+    } catch (e) {
+      return {
+        name: "pnpm",
+        shouldAddNoGitChecks: false
+      };
+    }
+  }
+  return {
+    name: "npm"
+  };
 }
 async function getTokenIsRequired() {
   const {
@@ -861,6 +871,10 @@ async function internalPublish(packageJson, opts, twoFactorState) {
       env: Object.assign({}, process.env, envOverride),
       cwd: opts.cwd,
       stdio: ["inherit", "inherit", "pipe"]
+    }) : publishTool.name === "bun" ? spawn$1("bun", ["publish", ...publishFlags], {
+      env: Object.assign({}, process.env, envOverride),
+      cwd: opts.publishDir,
+      stdio: ["inherit", "inherit", "pipe"]
     }) : spawn$1(publishTool.name, ["publish", opts.publishDir, ...publishFlags], {
       env: Object.assign({}, process.env, envOverride),
       stdio: ["inherit", "inherit", "pipe"]
@@ -905,6 +919,9 @@ async function internalPublish(packageJson, opts, twoFactorState) {
   } = publishTool.name === "pnpm" ? await spawn$1("pnpm", ["publish", ...publishFlags], {
     env: Object.assign({}, process.env, envOverride),
     cwd: opts.cwd
+  }) : publishTool.name === "bun" ? await spawn$1("bun", ["publish", ...publishFlags.filter(flag => flag !== "--json")], {
+    env: Object.assign({}, process.env, envOverride),
+    cwd: opts.publishDir
   }) : await spawn$1(publishTool.name, ["publish", opts.publishDir, ...publishFlags], {
     env: Object.assign({}, process.env, envOverride)
   });
